core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -1693,3 +1638,46 @@ private[spark] object JsonProtocol extends JsonUtils {
     }
   }
 }
+
+@deprecated("use new JsonProtocol(sparkConf) instead", "4.1.0")
+private[spark] object JsonProtocol extends JsonProtocol(new SparkConf()) {
```

Comment:
```suggestion
private[spark] object JsonProtocol {
```

Reply:
And the `@deprecated("use new JsonProtocol(sparkConf) instead", "4.1.0")` annotation can be removed. I have attempted to run the `dev/mima` command locally, and it executed successfully.

Reply:
Done